torch/nn/modules/module.py

```diff
@@ -1491,7 +1491,7 @@ def load(module, prefix=''):
                                self.__class__.__name__, "\n\t".join(error_msgs)))
         return _IncompatibleKeys(missing_keys, unexpected_keys)
 
-    def _named_members(self, get_members_fn, prefix='', recurse=True):
+    def _named_members(self, get_members_fn, prefix='', recurse=True, remove_duplicate=True):
         r"""Helper method for yielding various names + members of modules."""
         memo = set()
         modules = self.named_modules(prefix=prefix) if recurse else [(prefix, self)]
```

Comment:
You need to pass the remove_duplicate flag to the module traversal here no?

Reply:
Good catch - would be good to make sure the case of duplicate modules is tested as well